src/model.ts

```diff
@@ -462,3 +463,53 @@ export class SpecQueryModel {
     this._rankingScore[rankingName] = score;
   }
 }
+
+export class SpecQueryModelGroup {
+  private _name: string;
+  private _path: string;
+  private _items: (SpecQueryModel | SpecQueryModelGroup)[];
+  private _groupBy: GroupBy;
+  private _orderGroupBy: string;
+
+  constructor(name: string = '', path: string = '', items: (SpecQueryModel | SpecQueryModelGroup)[] = [],
+              groupBy: GroupBy = undefined, orderGroupBy: string = undefined) {
+    this._name = name;
+    this._path = path;
+    this._items = items;
+    this._groupBy = groupBy;
+    this._orderGroupBy = orderGroupBy;
+  }
+
+  public getTopSpecQueryModel(g: SpecQueryModelGroup): SpecQueryModel {
```

Comment:
Why should this accept g?   

It's now a class method.  Just use `this` instead of `g`.